eZ/Publish/SPI/Repository/Decorator/ContentServiceDecorator.php

```diff
@@ -0,0 +1,238 @@
+    }
+
+    public function loadContentListByContentInfo(
+        array $contentInfoList,
+        array $languages = [],
+        $useAlwaysAvailable = true
+    ) {
+        $this->innerService->loadContentListByContentInfo($contentInfoList, $languages, $useAlwaysAvailable);
+    }
+
+    public function createContent(
+        ContentCreateStruct $contentCreateStruct,
+        array $locationCreateStructs = []
+    ) {
+        $this->innerService->createContent($contentCreateStruct, $locationCreateStructs);
+    }
+
+    public function updateContentMetadata(
+        ContentInfo $contentInfo,
+        ContentMetadataUpdateStruct $contentMetadataUpdateStruct
+    ) {
+        $this->innerService->updateContentMetadata($contentInfo, $contentMetadataUpdateStruct);
+    }
+
+    public function deleteContent(ContentInfo $contentInfo)
+    {
+        $this->innerService->deleteContent($contentInfo);
+    }
+
+    public function createContentDraft(
+        ContentInfo $contentInfo,
+        VersionInfo $versionInfo = null,
+        User $creator = null
+    ) {
+        $this->innerService->createContentDraft($contentInfo, $versionInfo, $creator);
+    }
+
+    public function loadContentDrafts(User $user = null)
+    {
+        $this->innerService->loadContentDrafts($user);
+    }
+
+    public function updateContent(
+        VersionInfo $versionInfo,
+        ContentUpdateStruct $contentUpdateStruct
+    ) {
+        $this->innerService->updateContent($versionInfo, $contentUpdateStruct);
+    }
+
+    public function publishVersion(VersionInfo $versionInfo)
+    {
+        $this->innerService->publishVersion($versionInfo);
+    }
+
+    public function deleteVersion(VersionInfo $versionInfo)
+    {
+        $this->innerService->deleteVersion($versionInfo);
+    }
+
+    public function loadVersions(ContentInfo $contentInfo)
+    {
+        $this->innerService->loadVersions($contentInfo);
+    }
+
+    public function copyContent(
+        ContentInfo $contentInfo,
+        LocationCreateStruct $destinationLocationCreateStruct,
+        VersionInfo $versionInfo = null
+    ) {
+        $this->innerService->copyContent($contentInfo, $destinationLocationCreateStruct, $versionInfo);
+    }
+
+    public function loadRelations(VersionInfo $versionInfo)
+    {
+        $this->innerService->loadRelations($versionInfo);
+    }
+
+    public function loadReverseRelations(ContentInfo $contentInfo)
+    {
+        $this->innerService->loadReverseRelations($contentInfo);
+    }
+
+    public function addRelation(
+        VersionInfo $sourceVersion,
+        ContentInfo $destinationContent
+    ) {
+        $this->innerService->addRelation($sourceVersion, $destinationContent);
+    }
+
+    public function deleteRelation(
+        VersionInfo $sourceVersion,
+        ContentInfo $destinationContent
+    ) {
+        $this->innerService->deleteRelation($sourceVersion, $destinationContent);
+    }
+
+    public function removeTranslation(
```

Comment:
It's deprecated and will be removed shortly from 8.0, so there's no point in adding it now.

Reply:
It doesn't matter if it will be dropped shortly or not, because this layer have to be compatible with interface it implements, so till interface contains deprecated methods I have to support it as well.

Reply:
True